core/tests/puppeteer-acceptance-tests/spec/blog-admin-tests.spec.js

```diff
@@ -0,0 +1,77 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Acceptance Test for Blog Admin
+ */
+
+const userFactory = require(
+  '../puppeteer-testing-utilities/user-factory.js');
+const testConstants = require(
+  '../puppeteer-testing-utilities/test-constants.js');
+
+const DEFAULT_SPEC_TIMEOUT = testConstants.DEFAULT_SPEC_TIMEOUT;
+
+describe('Blog Admin', function() {
+  const ROLE_BLOG_ADMIN = 'blog admin';
+  const ROLE_BLOG_POST_EDITOR = 'blog post editor';
+  let superAdmin = null;
+  let blogAdmin = null;
+
+  beforeAll(async function() {
+    superAdmin = await userFactory.createNewSuperAdmin('superAdm');
+    blogAdmin = await userFactory.createNewBlogAdmin('blogAdm');
+  }, DEFAULT_SPEC_TIMEOUT);
+
+  /** TODO(#17162): This test should be done without the need of super admin, as
+  * blog admin must be able to revoke blog editor role of other users from the
+  * /blog-admin page. But this is not the case now, only super admin can do this
+  */
+  it('should assign roles to users and change tag properties',
+    async function() {
+      const guestUsr1 = await userFactory.createNewGuestUser(
+        'guestUsr1', 'guest_user1@example.com');
+      const guestUsr2 = await userFactory.createNewGuestUser(
+        'guestUsr2', 'guest_user2@example.com');
+
+      await superAdmin.expectUserNotToHaveRole('guestUsr1', ROLE_BLOG_ADMIN);
+      await blogAdmin.assignUserToRoleFromBlogAdminPage(
+        'guestUsr1', 'BLOG_ADMIN');
+      await superAdmin.expectUserToHaveRole('guestUsr1', ROLE_BLOG_ADMIN);
+
+      await superAdmin.expectUserNotToHaveRole(
+        'guestUsr2', ROLE_BLOG_POST_EDITOR);
+      await blogAdmin.assignUserToRoleFromBlogAdminPage(
+        'guestUsr2', 'BLOG_POST_EDITOR');
+      await superAdmin.expectUserToHaveRole('guestUsr2', ROLE_BLOG_POST_EDITOR);
+
+      await blogAdmin.removeBlogEditorRoleFromUsername('guestUsr2');
+      await superAdmin.expectUserNotToHaveRole(
+        'guestUsr2', ROLE_BLOG_POST_EDITOR);
+
+      await blogAdmin.expectTagToNotExistInBlogTags('Test_Tag');
+      await blogAdmin.addNewBlogTag('Test_Tag');
+      await blogAdmin.expectTagToExistInBlogTags('Test_Tag');
+
+      await blogAdmin.expectMaximumTagLimitNotToBe('5');
+      await blogAdmin.setMaximumTagLimitTo('5');
+      await blogAdmin.expectMaximumTagLimitToBe('5');
+      await guestUsr1.closeBrowser();
+      await guestUsr2.closeBrowser();
+    }, DEFAULT_SPEC_TIMEOUT);
```

Comment:
why do we need this timeout for each user story? by default, if any action in our test is taking more than 30sec, then it throws a timeout error (this is the default behavior in puppeteer). if the timeout for each action in the test is maintained, then bounding each test with explicit timeout would be correct or not?

Reply:
@Shivkant-Chauhan The timeout i added here is not in any way related to puppeteer tests. puppeteer tests timeout happends when some element we are expecting does'nt arrive in dom in expected time ( 30 sec here ) . However the DEFAULT_SPEC_TIMEOUT is entirely different. It's for how much time a test should take before it can be deemed as a failed test. So suppose we are running a test and it is taking like 2 min to completely run. This will outdo the default timeout of jasmine tests ( also 30 sec ) . Therefore it becomes compulsory we must define a timelimit underwhich a test will run completely. However there can be cases where we'll have tests running for over 5 min. There we'll again have to change the timeout constant. I hope this gives you some insight.

Reply:
okay.. got it! but can you clarify this: "This will outdo the default timeout of jasmine tests ( also 30 sec )" are you saying this will remove the default timeout checked by puppeteer for each action

Reply:
No. How can a variable defined in a Jasmine test structure alter puppeteer settings. The variable I am defining is for the spec ( it ) . So naturally it'll affect only Jasmine spec function related settings. I hope that clears it a bit. But the bottomline is that in no way it will affect puppeteer configuration

Reply:
okay..! so this timeout just bounds a time limit for each test, that ensures by any chance or bug, the test do not run infinitely on CI.